spark/src/main/scala/org/apache/spark/sql/delta/commands/DMLWithDeletionVectorsHelper.scala

```diff
@@ -391,14 +391,12 @@ object DeletionVectorBitmapGenerator {
       val filePathToDVDf = sparkSession.createDataset(filePathToDV)
 
       val joinExpr = filePathToDVDf("path") === matchedRowsDf(FILE_NAME_COL)
-      val joinedDf = matchedRowsDf.join(filePathToDVDf, joinExpr)
-      assert(joinedDf.count() == matchedRowsDf.count(),
-        s"""
-           |The joined DataFrame should contain the same number of entries as the original
-           |DataFrame. It is likely that _metadata.file_path is not encoded by Spark as expected.
-           |Joined DataFrame count: ${joinedDf.count()}
-           |matchedRowsDf count: ${matchedRowsDf.count()}
-           |""".stripMargin)
+      // Perform leftOuter join to make sure we do not eliminate any rows because of path
+      // encoding issues. If there is such an issue we will detect it during the aggregation
+      // of the bitmaps.
+      val joinedDf = matchedRowsDf.join(filePathToDVDf, joinExpr, "leftOuter")
```

Comment:
The sanity wasn't created only because of Spark 3.4.0, and bug was fixed in 3.4.1/3.5.0? Can't we just remove the assert?

Reply:
Thanks for the review @felipepessoto. We fixed related issues related to the sanity check in Spark 3.5. In any case, the sanity check is very cheap now and it is good to have a safeguard in case we come across any issues in the future.

Reply:
Oh ok, the join is supposed to always match, so we are not returning more rows than before.

Reply:
Exactly :).